test/Popup.test.js

```diff
@@ -0,0 +1,146 @@
+      var popup = new Popup(this.container);
+      popup.setPosition(undefined, undefined);
+      assert(isNaN(popup.x));
+      assert(isNaN(popup.y));
+    });
+  });
+
+  describe('setText', function () {
+
+    it('using Element replaces innerHTML', function () {
+      var popup = new Popup(this.container);
+      popup.frame.innerHTML = '<div>This will get cleared!</div>';
+      popup.setText(document.createElement('div'));
+      assert.equal(popup.frame.innerHTML, '<div></div>');
+    });
+
+    it('using string replaces innerHTML', function () {
+      var popup = new Popup(this.container);
+      popup.frame.innerHTML = '<div>This will get cleared!</div>';
+      popup.setText('your text here!');
+      assert.equal(popup.frame.innerHTML, 'your text here!');
+    });
+  });
+
+  describe('show', function () {
+
+    it('set to undefined will show', function () {
+      var popup = new Popup(this.container);
+      popup.show(undefined);
+      assert.equal(popup.hidden, false);
+      assert.notEqual(popup.frame.style.left, "0px");
+      assert.notEqual(popup.frame.style.top, "0px");
+      assert.equal(popup.frame.style.visibility, "visible");
+    });
+
+    it('set to true will show', function () {
+      var popup = new Popup(this.container);
+      popup.show(true);
+      assert.equal(popup.hidden, false);
+      assert.notEqual(popup.frame.style.left, "0px");
+      assert.notEqual(popup.frame.style.top, "0px");
+      assert.equal(popup.frame.style.visibility, "visible");
+    });
+
+    it('set to true with overflowMethod "flip" will show', function () {
+      var popup = new Popup(this.container, 'flip');
+      popup.show(true);
+      assert.equal(popup.hidden, false);
+      assert.equal(popup.frame.style.left, "0px");
+      assert.equal(popup.frame.style.top, "0px");
+      assert.equal(popup.frame.style.visibility, "visible");
+    });
+
+    it('set to false will hide', function () {
+      var popup = new Popup(this.container);
+      popup.show(false);
+      assert.equal(popup.hidden, true);
+      assert.equal(popup.frame.style.left, "0px");
+      assert.equal(popup.frame.style.top, "0px");
+      assert.equal(popup.frame.style.visibility, "hidden");
+    });
+  });
+
+  describe('hide', function () {
+
+    it('sets hidden to true, frame style to 0,0 and visibility to hidden', function () {
+      var popup = new Popup(this.container);
+      popup.hide();
+      assert.equal(popup.hidden, true);
+      assert.equal(popup.frame.style.left, "0px");
+      assert.equal(popup.frame.style.top, "0px");
+      assert.equal(popup.frame.style.visibility, "hidden");
+    });
+  });
+
+  describe('destroy', function () {
+
+    it('removes frame from container', function () {
+      assert.equal(this.container.children.length, 0);
+      var popup = new Popup(this.container);
+      assert.equal(this.container.children.length, 1);
+      popup.destroy();
+      assert.equal(this.container.children.length, 0);
```

Comment:
Although I disagree with this from a purist standpoint, I'll concede to practicality.